test/performance/benchmarks/dataplane-probe/main.go

```diff
@@ -0,0 +1,181 @@
+			// on the dataplane, and so it is intended as a canary to flag environmental
+			// problems that might be causing contemporaneous Knative or Istio runs to fall out of SLA.
+			slaMin: minDefault,
+			slaMax: 105 * time.Millisecond,
+		},
+		"queue": {
+			target: vegeta.Target{
+				Method: http.MethodGet,
+				URL:    "http://queue-proxy.default.svc.cluster.local?sleep=100",
+			},
+			// hitting a Knative Service
+			// going through JUST the queue-proxy falls in the +10ms range.
+			slaMin: minDefault,
+			slaMax: 110 * time.Millisecond,
+		},
+		"activator": {
+			target: vegeta.Target{
+				Method: http.MethodGet,
+				URL:    "http://activator.default.svc.cluster.local?sleep=100",
+			},
+			// hitting a Knative Service
+			// going through BOTH the activator and queue-proxy falls in the +10ms range.
+			slaMin: minDefault,
+			slaMax: 110 * time.Millisecond,
+		},
+	}
+)
+
+func main() {
+	ctx := signals.NewContext()
+	cfg := injection.ParseAndGetRESTConfigOrDie()
+	ctx, startInformers := injection.EnableInjectionOrDie(ctx, cfg)
+	startInformers()
+
+	if *target == "" {
+		log.Fatalf("-target is a required flag.")
+	}
+
+	log.Println("Starting dataplane probe for target:", *target)
+
+	ctx, cancel := context.WithTimeout(ctx, *duration+time.Minute)
+	defer cancel()
+
+	// Based on the "target" flag, load up our target benchmark.
+	// We only run one variation per run to avoid the runs being noisy neighbors,
+	// which in early iterations of the benchmark resulted in latency bleeding
+	// across the different workload types.
+	t, ok := targets[*target]
+	if !ok {
+		log.Fatalf("Unrecognized target: %s", *target)
+	}
+
+	// Make sure the target is ready before sending the large amount of requests.
+	if err := performance.ProbeTargetTillReady(t.target.URL, *duration); err != nil {
+		log.Fatalf("Failed to get target ready for attacking: %v", err)
+	}
+
+	// Send 1000 QPS (1 per ms) for the given duration with a 30s request timeout.
+	rate := vegeta.Rate{Freq: 1, Per: time.Millisecond}
+	targeter := vegeta.NewStaticTargeter(t.target)
+	attacker := vegeta.NewAttacker(vegeta.Timeout(30 * time.Second))
+
+	influxReporter, err := performance.NewInfluxReporter(map[string]string{"target": *target})
+	if err != nil {
+		log.Fatalf("failed to create influx reporter: %v", err.Error())
+	}
+	defer influxReporter.FlushAndShutdown()
+
+	// Start the attack!
+	results := attacker.Attack(targeter, rate, *duration, "load-test")
+	deploymentStatus := performance.FetchDeploymentStatus(ctx, system.Namespace(), "activator", time.Second)
+
+	metricResults := &vegeta.Metrics{}
+
+LOOP:
+	for {
+		select {
+		case <-ctx.Done():
+			// If we time out or the pod gets shutdown via SIGTERM then start to
+			// clean thing up.
+			break LOOP
+
+		case ds := <-deploymentStatus:
+			// Report number of ready activators.
+			influxReporter.AddDataPoint(benchmarkName, map[string]interface{}{"activator-pod-count": float64(ds.ReadyReplicas)})
```

Comment:
Should not this be uint64? Could this be a float?

Reply:
I think this even can be an int32. I don't expect that many replicas to ever be created.